src/Variant.h

```diff
@@ -11,33 +11,42 @@ using boost::bad_lexical_cast;
 
 enum class DataTypes
 {
-    NULL_T, BOOL, INT, FLOAT, STRING
+    NULL_T, BOOL, INT, DOUBLE, STRING
 };
 
 
 class Variant
 {
 public:
-    Variant() :                 b_(bool()), i_(int()), f_(float()), datatype(DataTypes::NULL_T) {}
-    explicit Variant(bool b) :  b_(b),      i_(int()), f_(float()), datatype(DataTypes::BOOL)   {}
-    explicit Variant(int i) :   b_(bool()), i_(i),     f_(float()), datatype(DataTypes::INT)    {}
-    explicit Variant(float f) : b_(bool()), i_(int()), f_(f),       datatype(DataTypes::FLOAT)  {}
-    explicit Variant(const std::string& s) : b_(bool()), i_(int()), f_(float()), s_(s), datatype(DataTypes::STRING) {}
-
-    bool toBool();
-    int toInt();
-    float toFloat();
-    std::string toString();
+    Variant() :                  b_(nullptr),     i_(nullptr), d_(nullptr), s_(nullptr), datatype(DataTypes::NULL_T)   {}
```

Comment:
As for nullptr you can use boost::optional. It contains some variable or nothing and you can check it. But dont use pointers